### PR TITLE
Updates pac4j to implement new Google's People API

### LINKED ISSUE #1511: Google OAuth couldn't be used for new projects due to legacy API still in use
Google OAuth authentication in the WebAPI uses Legacy People API to retreive info about person who has logged in.
But this API is [deprecated](https://developers.google.com/people/legacy) since March 7, 2019 and cannot be used in new projects. This means none is able to deploy new WebAPI istance with Google OAuth authentication.
Google recommends migrate to the new People API.

### PR DESCRIPTION
**Pull request by wivern**
Fixes #1511

### CODE REVIEW

**Comment by anton-abushkevich** [comment]
when trying to authenticate with Google oAuth:
```2020-04-16 19:03:43.674 ERROR http-nio-8080-exec-8 org.ohdsi.webapi.shiro.filters.ExceptionHandlerFilter -  - Error during filtering
javax.servlet.ServletException: org.pac4j.core.exception.TechnicalException: clientId cannot be blank
	at org.apache.shiro.web.servlet.AdviceFilter.cleanup(AdviceFilter.java:196)
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:148)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:118)
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108)
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137)
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.ohdsi.webapi.shiro.filters.CacheFilter.doFilter(CacheFilter.java:30)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.ohdsi.webapi.shiro.filters.ExceptionHandlerFilter.doFilter(ExceptionHandlerFilter.java:28)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:109)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.pac4j.core.exception.TechnicalException: clientId cannot be blank
	at org.pac4j.core.util.CommonHelper.assertTrue(CommonHelper.java:107)
	at org.pac4j.core.util.CommonHelper.assertNotBlank(CommonHelper.java:119)
	at org.pac4j.core.util.CommonHelper.assertNotBlank(CommonHelper.java:129)
	at org.pac4j.oidc.config.OidcConfiguration.internalInit(OidcConfiguration.java:124)
	at org.pac4j.core.util.InitializableObject.init(InitializableObject.java:20)
	at org.pac4j.oidc.client.OidcClient.clientInit(OidcClient.java:50)
	at org.pac4j.core.client.IndirectClient.internalInit(IndirectClient.java:63)
	at org.pac4j.core.util.InitializableObject.init(InitializableObject.java:20)
	at org.pac4j.core.client.finder.DefaultCallbackClientFinder.find(DefaultCallbackClientFinder.java:38)
	at org.pac4j.core.engine.DefaultCallbackLogic.perform(DefaultCallbackLogic.java:73)
	at io.buji.pac4j.filter.CallbackFilter.doFilter(CallbackFilter.java:76)
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108)
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137)
	... 56 more
```

File: src/main/java/org/pac4j/oidc/profile/converter/OidcLongTimeConverter.java [hunk 1]
```diff
@@ -34,18 +32,18 @@ public OidcLongTimeConverter() {
 
     public Date convert(Object attribute) {
         if (attribute instanceof Long) {
-            long seconds = ((Long)attribute).longValue();
-            return new FormattedDate(new Date(seconds * 1000L), "yyyy-MM-dd'T'HH:mm:ss'z'", Locale.getDefault());
+            long seconds = (Long) attribute;
+            return new Date(seconds * 1000L);
```
Comment (chrisknoll) [inline src/main/java/org/pac4j/oidc/profile/converter/OidcLongTimeConverter.java:36]: Does the `seconds * 1000L` mean that these are actually milliseconds?  Maybe rename the variable to reflect this?

**Comment by chrisknoll** [comment]
I'm attempting to pull in the commits for his PR into 2.8 because other changes are requested that depends on a new pac4j library, and I ran into this issue during compile:

  bad class file: C:\Users\cknoll1\.m2\repository\org\opensaml\opensaml-saml-api\4.0.0\opensaml-saml-api-4.0.0.jar(org/opensaml/saml/common/xml/SAMLConstants.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.

I'm assuming this dependnecy came from the update, and it appears that this dependency requires JDK 11.  Am I reading that correctly?

**Comment by wivern** [comment]
@chrisknoll Yes, it seems you're right with the java version. 
I guess for java 8 this dependency should be used https://mvnrepository.com/artifact/org.pac4j/pac4j-saml-opensamlv3/4.0.0

**Comment by chrisknoll** [comment]
Ok, I will give it a shot, and we'll just be aware of it when we merge other 2.7.7 updates into 2.8

**Comment by chrisknoll** [comment]
I'm struggling a little bit to reconcile these API changes.  After pulling the commits into 2.8, there's compiler errors with this line:
```
       Resource keystorePath = CommonHelper.getResource(keyStoreFile);
```

The getResource() doesn't exist anymore on CommonHelper.  But I would have expected the updated commits for pac4j v4.0 to acocunt for that, but it did not.  Also, `mport org.pac4j.saml.client.SAML2ClientConfiguration;` is not resolving.

Any information on these 2 errors would be helpful.

**Comment by wivern** [comment]
The `keystorePath` is used by the `SAML2Configuration` and the latest version of pac4j-saml uses Spring's Resource class or just a string with filepath. So `CommonHelper` isn't required any more.

And the `SAML2Configuration` was moved to the new package `org.pac4j.saml.config`

**Comment by chrisknoll** [comment]
Ok, thanks.  I finally figured that out too.  I have a branch that incorporates the pac4j upates with the SAML changes tot he api, but I noticed other differences (ie: exception handling), so, i'm about to push up that branch as a PR, and if you could pull it and try it and see if I missed anything. It's possible that the cherry picks that were made from the 2.7.7 branch may not have had everything i needed to get a complete feature, but the problem is 2 fold: the new features were spread across a few commits and the hot-fix releases (2.7.7) had new features!  Trying to avoid those problems in the future...

**Pull request merged**